src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,101 @@
+import {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+
+import { Person } from '../../types/Person';
+import { PersonList } from '../PersonList';
+import './Autocomplete.scss';
+
+type Props = {
+  people: Person[],
+  onSelect: (person: Person) => void,
+  delay: number,
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  onSelect,
+  delay,
+}) => {
+  const [isListVisible, setIsListVisible] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const [debauncedInputValue, setDebauncedInputValue] = useState(inputValue);
+  const [wasThereClick, setWasThereClick] = useState(true);
+
+  const handleInputFocus = (event: React.FocusEvent<HTMLInputElement>) => {
+    if (event.target.value === '') {
+      setIsListVisible(true);
+      setWasThereClick(false);
+    }
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+    setIsListVisible(false);
+    setWasThereClick(false);
+  };
+
+  const handleItemClick = useCallback((person: Person) => {
+    onSelect(person);
+    setInputValue(person.name);
+    setIsListVisible(false);
+    setWasThereClick(true);
+  }, []);
```

Comment:
The callback functions in the useCallback hook are missing its dependency array. The current implementation may lead to bugs because it will not capture changes in the onSelect prop, and always use the initial value of onSelect and other

Reply:
here, when I tried to add person into dependencies, I got a message about mistake that 'person can not be find' because this variable located only inside this function